application/src/main/java/com/knubisoft/application/ecryption/impl/EncryptionImpl.java

```diff
@@ -0,0 +1,54 @@
+package com.knubisoft.application.ecryption.impl;
+
+import com.knubisoft.application.ecryption.Encryption;
+import lombok.SneakyThrows;
+import org.bouncycastle.cms.CMSAlgorithm;
+import org.bouncycastle.cms.CMSEnvelopedData;
+import org.bouncycastle.cms.CMSEnvelopedDataGenerator;
+import org.bouncycastle.cms.CMSProcessableByteArray;
+import org.bouncycastle.cms.CMSTypedData;
+import org.bouncycastle.cms.KeyTransRecipientInformation;
+import org.bouncycastle.cms.RecipientInformation;
+import org.bouncycastle.cms.jcajce.JceCMSContentEncryptorBuilder;
+import org.bouncycastle.cms.jcajce.JceKeyTransEnvelopedRecipient;
+import org.bouncycastle.cms.jcajce.JceKeyTransRecipient;
+import org.bouncycastle.cms.jcajce.JceKeyTransRecipientInfoGenerator;
+import org.bouncycastle.operator.OutputEncryptor;
+
+import java.security.PrivateKey;
+import java.security.cert.X509Certificate;
+import java.util.Collection;
+
+import static java.util.Objects.nonNull;
+
+public class EncryptionImpl implements Encryption {
+
+    @Override
+    @SneakyThrows
+    public byte[] encryptData(byte[] data, X509Certificate encryptionCertificate) {
+        if (nonNull(data) && nonNull(encryptionCertificate)) {
```

Comment:
If data or encryptionCertificate are nulls - throw exception pls. The same for decrypt method